tests/functional/tests.py

```diff
@@ -343,92 +668,231 @@ def test_presigned_put_object(client):
+        log_output['duration'] = duration = round(time.time() - start_time, 2)
+        print_json(log_output)
+        exit()
     finally:
-        client.remove_bucket(bucket_name)
+        try:
+            client.remove_bucket(bucket_name)
+        except Exception as err:
+            log_output['message'] = err
+            log_output['error'] = traceback.format_exc()
+            log_output['status'] = FAIL
+            log_output['duration'] = duration = round(time.time() - start_time, 2)
+            print_json(log_output)
+            exit()
+    # Test passes
+    log_output['duration'] = duration = round(time.time() - start_time, 2)
+    print_json(log_output)
 
 def test_set_bucket_policy(client):
-    bucket_name = uuid.uuid4().__str__()
+    start_time = time.time()
+
+    log_output = new_log_result(client.set_bucket_policy)
+    log_output['description'] = 'Tests set_bucket_policy api'
+    # Get a unique bucket_name
+    log_output['args']['bucket_name'] = bucket_name = uuid.uuid4().__str__()
+    log_output['args']['prefix'] = prefix = '1/'
     try:
         client.make_bucket(bucket_name)
         # Set read-only policy successfully.
-        client.set_bucket_policy(bucket_name, '1/', Policy.READ_ONLY)
+        client.set_bucket_policy(bucket_name, prefix, Policy.READ_ONLY)
         # Set read-write policy successfully.
-        client.set_bucket_policy(bucket_name, '1/', Policy.READ_WRITE)
+        client.set_bucket_policy(bucket_name, prefix, Policy.READ_WRITE)
         # Reset policy to NONE.
-        client.set_bucket_policy(bucket_name, '', Policy.NONE)
+        # Added into log output for clarity/debugging purposes
+        log_output['args']['prefix-2'] = prefix = ''
+        client.set_bucket_policy(bucket_name, prefix, Policy.NONE)
         # Validate if the policy is reverted back to NONE.
         policy_name = client.get_bucket_policy(bucket_name)
         if policy_name != Policy.NONE:
             raise ValueError('Policy name is invalid ' + policy_name)
+    except Exception as err:
+        log_output['message'] = err
+        log_output['error'] = traceback.format_exc()
+        log_output['status'] = 'FAIL1'
+        log_output['duration'] = duration = round(time.time() - start_time, 2)
+        print_json(log_output)
+        exit()
     finally:
-        client.remove_bucket(bucket_name)
+        try:
+            client.remove_bucket(bucket_name)
+        except Exception as err:
+            log_output['message'] = err
+            log_output['error'] = traceback.format_exc()
+            log_output['status'] = 'FAIL2'
+            log_output['duration'] = duration = round(time.time() - start_time, 2)
+            print_json(log_output)
+            exit()
+    # Test passes
+    log_output['duration'] = duration = round(time.time() - start_time, 2)
+    print_json(log_output)
 
 def test_remove_objects(client):
-    bucket_name = uuid.uuid4().__str__()
+    start_time = time.time()
```

Comment:
In another PR, when a class will be introduced, we can make new_log_result() internally stores start time,